applications/stream-applications-core/stream-applications-security-common/src/main/java/org/springframework/cloud/stream/app/security/common/AppStarterWebSecurityAutoConfiguration.java

```diff
@@ -47,29 +59,71 @@
 @EnableWebSecurity
 public class AppStarterWebSecurityAutoConfiguration {
 
+	@Autowired
+	private WebEndpointProperties webEndpointProperties;
+
+	/*
+	 * Spring security will discover this provider. It grants the ADMIN ROLE to the admin
+	 * user, if configured, otherwise it passes through the current authentication.
+	 */
+	@Bean
+	AuthenticationProvider adminAuthenticationProvider(
+			AppStarterWebSecurityAutoConfigurationProperties securityProperties) {
+		return new AuthenticationProvider() {
+			@Override
+			public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+				if (securityProperties.isEnabled() &&
+						StringUtils.hasText(securityProperties.getAdminUser()) &&
+						securityProperties.getAdminUser().equals(authentication.getName()) &&
+						StringUtils.hasText(securityProperties.getAdminPassword()) &&
+						securityProperties.getAdminPassword().equals(authentication.getCredentials().toString())) {
```

Comment:
Will `credentials` always be non-null?

Reply:
I believe that's the case once it gets here, but I'll add a check.